js/hotgraphicView.js

```diff
@@ -0,0 +1,223 @@
+                this.replaceWithNarrative();
+            }
+        },
+
+        inview: function(event, visible, visiblePartX, visiblePartY) {
+            if (visible) {
+                if (visiblePartY === 'top') {
+                    this._isVisibleTop = true;
+                } else if (visiblePartY === 'bottom') {
+                    this._isVisibleBottom = true;
+                } else {
+                    this._isVisibleTop = true;
+                    this._isVisibleBottom = true;
+                }
+
+                if (this._isVisibleTop && this._isVisibleBottom) {
+                    this.$('.component-widget').off('inview');
+                    this.setCompletionStatus();
+                }
+            }
+        },
+
+        replaceWithNarrative: function() {
+            if (!Adapt.componentStore.narrative) throw "Narrative not included in build";
+            var NarrativeView = Adapt.componentStore.narrative.view;
+
+            var model = this.prepareNarrativeModel();
+            var newNarrative = new NarrativeView({ model: model });
+            var $container = $(".component-container", $("." + this.model.get("_parentId")));
+
+            this.model.set('_isPopupOpen', false); // close popup 
+
+            newNarrative.reRender();
+            newNarrative.setupNarrative();
+            $container.append(newNarrative.$el);
+            Adapt.trigger('device:resize');
+            _.defer(_.bind(function () {
+                this.remove();
+            }, this));
+        },
+
+        prepareNarrativeModel: function() {
+            var model = this.model;
+            model.set({
+                '_component': 'narrative',
+                '_wasHotgraphic': true,
+                'originalBody': model.get('body'),
+                'originalInstruction': model.get('instruction')
+            });
+
+            // check if active item exists, default to 0
+            var active = model.getActiveItem();
+            if (!active) {
+                model.getItem(0).toggleActive(true);
+            }
+            
+            if (model.get('mobileBody')) {
+                model.set('body', model.get('mobileBody'));
+            }
+            if (model.get('mobileInstruction')) {
+                model.set('instruction', model.get('mobileInstruction'));
+            }
+            return model;
+        },
+
+        onItemsActiveChange: function(model, _isActive) {
+            var selector = 'item-'+model.get('_index');
+            this.$('.hotgraphic-graphic-pin.'+selector).toggleClass('active', _isActive);
+        },
+        
+        onItemsVisitedChange: function(model, _isVisited) {
+            if (_isVisited) {
+                var selector = 'item-'+model.get('_index');
+                this.$('.hotgraphic-graphic-pin.'+selector).addClass('visited');
+            }
+        },
+
+        onPinClicked: function (event) {
+            if(event) {
```

Comment:
`if (event) event.preventDefault();` would be fine for these but leave it as-is if you want!